### ISSUE TITLE
React's state is being mutated

### ISSUE THREAD

**Issue by vitalybe** [krawaller/reactexperiment#1]
Per React's docs, the state [should not be mutated](https://facebook.github.io/react/docs/component-api.html). 

The following [code lines](https://github.com/krawaller/reactexperiment/blob/gh-pages/src/board.jsx#L20-52), however, do mutate it. Consider this line: `tilestates: _.extend(this.state.tilestates,_.object([index],['revealed']))`

In this line, `_.extend` would change the property of this.state.

The following StackOverflow questions looks into this issue:
http://stackoverflow.com/questions/26253351/correct-modification-of-state-arrays-in-reactjs
http://stackoverflow.com/questions/25008322/updating-deep-reactjs-state?lq=1

Basically to do it correctly (but inefficiently) would be to use a deep clone and perform the change on it. The best way would probably be to use [Immutability Helpers](https://facebook.github.io/react/docs/update.html) or other similar tools.
